OMCompiler/Compiler/FrontEnd/DAE.mo

```diff
@@ -269,7 +269,17 @@ public uniontype Element
     ElementSource source "the origin of the component/equation/algorithm" ;
   end WHEN_EQUATION;
 
-  record FOR_EQUATION " a for-equation"
+  record INITIAL_FOR_EQUATION " a for-equation"
+    Type type_ "this is the type of the iterator";
+    Boolean iterIsArray "True if the iterator has an array type, otherwise false.";
+    Ident iter "the iterator variable";
+    Integer index "the index of the iterator variable, to make it unique; used by the new inst";
+    Exp range "range for the loop";
+    list<Element> equations "Equations" ;
+    ElementSource source "the origin of the component/equation/algorithm" ;
+  end INITIAL_FOR_EQUATION;
+
+  record FOR_EQUATION " an initial for-equation"
```

Comment:
Typo :slightly_smiling_face: 
```suggestion
  record FOR_EQUATION " a for-equation"
```

BTW why don't we have a boolean for this? @kabdelhak what do you think?

Reply:
Never mind, I found the function `isInitial`
https://github.com/kabdelhak/OpenModelica/blob/9b17a155fb3c49a64aa79dd9ad0412ba2f7efa05/OMCompiler/Compiler/NBackEnd/Classes/NBEquation.mo#L1088

Reply:
Just checking that someone is paying attention :smile: Fixed in #8208.

And yeah, we could have used booleans instead of separate records for the initial versions. But I wanted to keep it consistent with how all the other equations are defined in the DAE, and changing the old records would be a lot of work. The new backend doesn't use these at all I think, so it should go away in the future.